### PR TITLE
Add protection for admin page using Rack simple auth

### PR DESCRIPTION
**Pull request by Bounga**
Issue #39 is requesting to add a basic auth for admin app. This PR adds simple auth for an admin page that is dedicated to become the page that lists all the projects available.

It tries to find env variable provided on app boot or uses hanami / hanami as default.

This PR adds tests to ensure access is avoided without providing credential or even wrong credentials. It also add a test that ensures it works with valid credentials.

Hope it fits the requirements.

### CODE REVIEW

**Comment by cllns** [comment]
Thanks! I know you were just following @davydovanton's request but I'd personally rather have _no_ default authentication. We should be secure by default, not insecure by default. If we use `ENV.fetch` with no second param, there will be a `KeyError` alerting the user that there's no Username and Password. (I'd also be OK with a default user name but not a default password, that's the important one for me.)

**Comment by Bounga** [comment]
@cllns I'm 100% ok with this. I was pretty uncomfortable with the default values. It's a big security hole to me but hey @davydovanton is kind of my boss here. I'd be happy to remove the default password to raise an exception if not provided as an env var on boot.

**Comment by davydovanton** [comment]
agree with @cllns here. Let's put ENV variables to `.env.development` and `.env.test` files and raise error if variables doesn't set :)

**Comment by Bounga** [comment]
Using `fetch` in `admin/application.rb` `controller.prepare` block will raise the error at runtime. I mean when we try to access the admin pages and not when the server is started. Doesn't sound great to me. 

I'd like to move the `fetch` calls at the beginning of the `configure` block and store it in constants so it will raise as soon as you start the app rather than only when trying to access the admin pages.

What do you think about it?

**Comment by cllns** [comment]
I like that :+1: Great idea. (Kind of goes against Ruby being dynamic, but for a good reason, I think)

**Comment by Bounga** [comment]
Here is the modified version.

File: apps/admin/application.rb [hunk 1]
```diff
@@ -258,9 +258,16 @@ class Application < Hanami::Application
       # This is useful for sharing common functionality
       #
       # See: http://www.rubydoc.info/gems/hanami-controller#Configuration
+      ADMIN_USERNAME = ENV.fetch('ADMIN_USERNAME')
+      ADMIN_PASSWORD = ENV.fetch('ADMIN_PASSWORD')
```
Comment (davydovanton) [inline apps/admin/application.rb]: good idea, WDYT if we replace it to initialize file? (I can do it after merge :) )

Comment (Bounga) [reply apps/admin/application.rb]: Yes that's a very good idea 👍.

**Comment by davydovanton** [comment]
thanks for contribution, it's awesome work! 💯

**Comment by Bounga** [comment]
@davydovanton Does CI deploys the website automatically when something is merged in master because http://contributors.hanamirb.org returns a 500. Coincidence or the CI just didn't set the required env variables for the admin pages?

**Pull request merged**